lynx/src/main/java/com/github/pedrovgs/lynx/model/Lynx.java

```diff
@@ -143,10 +161,17 @@ private boolean shouldAddTrace(String logcatTrace) {
   }
 
   private synchronized boolean traceMatchesFilter(String logcatTrace) {
-    TraceLevel levelFilter = lynxConfig.getFilterTraceLevel();
-    String filter = lynxConfig.getFilter().toLowerCase();
-    String logcatTraceLowercase = logcatTrace.toLowerCase();
-    return logcatTraceLowercase.contains(filter) && containsTraceLevel(logcatTrace, levelFilter);
+    return traceStringMatchesFilter(logcatTrace)
+            && containsTraceLevel(logcatTrace, lynxConfig.getFilterTraceLevel());
+  }
+
+  private boolean traceStringMatchesFilter(String logcatTrace) {
+    String lowerCaseLogcatTrace = logcatTrace.toLowerCase();
+    boolean matchesFilter = lowerCaseLogcatTrace.contains(lowerCaseFilter);
+    if (!matchesFilter && regexpFilter != null) {
```

Comment:
I did intentionally in that way, so if there's people already using the library with filters that corresponds to a regexp but they just want to match that text as, filter would continue applying. For me both approaches are ok :) After knowing that, let me know what do you prefer and I'll change it ;) Thanks!